ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh

```diff
@@ -29,46 +29,123 @@ then
   gcloud config set project "${GOOGLE_PROJECT_ID}"
 fi
 
+validation_result_file=$(mktemp)
+
 # User-defined tags validation. It will check if each user-defined tag is applied. 
 # Return non-zero is one or more user-defined tag absent. 
 # $1 - the current tags of the resource under question
 function validate_user_tags() {
   local -r current_tags="$1";  shift
 
-  local ret=0 cnt=1 a_tag_value
+  local cnt=1 a_tag_value
+  echo "" > "${validation_result_file}"
   printf '%s' "${USER_TAGS:-}" | while read -r PARENT KEY VALUE || [ -n "${PARENT}" ]
   do
     a_tag_value="namespacedTagValue: ${PARENT}/${KEY}/${VALUE}"
     if echo "${current_tags}" | grep -Fq "${a_tag_value}"; then
       echo "$(date -u --rfc-3339=seconds) - Found tag ${cnt} '${a_tag_value}' (PARENT/KEY/VALUE)."
       cnt=$(( $cnt + 1 ))
+      echo 0 >> "${validation_result_file}"
       continue
     else
       echo "$(date -u --rfc-3339=seconds) - Failed to find tag '${a_tag_value}' (PARENT/KEY/VALUE)."
-      ret=1
+      echo 1 >> "${validation_result_file}"
     fi
   done
-  return $ret
+}
+
+# check if OCP version will be equal to or greater than the minimum version
+# $1 - the minimum version to be compared with
+# return 0 if OCP version >= the minimum version, otherwise 1
+function version_check() {
+  local -r minimum_version="$1"
+  local ret
+
+  dir=$(mktemp -d)
+  pushd "${dir}"
+
+  cp ${CLUSTER_PROFILE_DIR}/pull-secret pull-secret
+  KUBECONFIG="" oc registry login --to pull-secret
+  ocp_version=$(oc adm release info --registry-config pull-secret ${TESTING_RELEASE_IMAGE} --output=json | jq -r '.metadata.version' | cut -d. -f 1,2)
+
+  echo "[DEBUG] minimum OCP version: '${minimum_version}'"
+  echo "[DEBUG] current OCP version: '${ocp_version}'"
+  curr_x=$(echo "${ocp_version}" | cut -d. -f1)
+  curr_y=$(echo "${ocp_version}" | cut -d. -f2)
+  min_x=$(echo "${minimum_version}" | cut -d. -f1)
+  min_y=$(echo "${minimum_version}" | cut -d. -f2)
+
+  if [ ${curr_x} -gt ${min_x} ] || ([${curr_x} -eq ${min_x} ] && [ ${curr_y} -ge ${min_y} ]); then
```

Comment:
I think we even do not need to care about `curr_x`, just comparing `curr_y` and `min_y` should be enough.

Reply:
But what if comparing "4.17" with "5.1" for example? Thanks!

Reply:
Do we have 5.X ocp version ?

Reply:
No for now, but I think the function should support comparing 4.m with 5.n. WDYT? Thanks!

Reply:
It is too far when we get OCP 5.x,  if we have 5.x, that sounds a different product, I guess most of steps can not be used created for 4.x.